SSA/Projects/InstCombine/HackersDelight.lean

```diff
@@ -9,27 +9,55 @@ namespace HackersDelight
 
 namespace Ch2Basics
 
-theorem neg_add_eq_neg_sub {x: BitVec w } :
-    -(x + y) = -x - y := by
-  ring
-
-theorem neg_eq_not_add_one {x : BitVec w} :
-    -x = ~~~x + 1 := by
-  exact BitVec.neg_eq_not_add x
-
 theorem not_eq_neg_sub_one {x : BitVec w} :
     ~~~ x = -x - 1 := by
   apply BitVec.eq_sub_iff_add_eq.mpr
   rw [BitVec.neg_eq_not_add]
 
+theorem not_and_eq_not_or_not {x y : BitVec w} :
+    ~~~ (x &&& y) = ~~~ x ||| ~~~ y := by
+  sorry
+
+theorem not_or_eq_not_and_not {x y : BitVec w} :
+    ~~~ (x ||| y) = ~~~ x &&& ~~~ y := by
+  sorry
+
+theorem not_add_eq_not_sub_one {x : BitVec w} :
+    ~~~ (x + 1) = ~~~ x - 1 := by
+  repeat rw [not_eq_neg_sub_one]
```

Comment:
Semantically, it's the same, but you may prefer `simp only` to `repeat rw`, because the former tactic will throw an error if it makes no progress.

```suggestion
  simp only [not_eq_neg_sub_one]
```